documentation.md

```diff
@@ -0,0 +1,97 @@
+in order to use Turbo Drive, Frames and Streams in our app.
+
+### Model Layer
+
+Throughout these docs we'll be using these two basic models for a chat app, similar to the
+[Hotwire demo video](https://www.youtube.com/watch?v=eKY-QES1XQQ):
+
+```python
+# chat/models.py
+
+from django.db import models
+class Room(models.Model):
+    name = models.CharField(max_length=255)
+
+
+class Message(models.Model):
+    room = models.ForeignKey(Room, related_name="messages", on_delete=models.CASCADE)
+    text = models.TextField()
+```
+
+`turbo-django` provides a `BroadcastableMixin` that enables CRUD operations from a given model
+to be broadcast over websockets. To enable messages to be broadcast over websocket to users
+listening in on a particular room, we add the `BroadcastableMixin` and tell `turbo-django`
+that actions on messages should be broadcast to the related `Room` as well:
+
+```python
+class Message(BroadcastableMixin, models.Model):
+    broadcast_to = ["room"]
+    broadcast_self = False
+
+    room = models.ForeignKey(Room, related_name="messages", on_delete=models.CASCADE)
+    text = models.TextField()
+    created_at = models.DateTimeField(auto_now_add=True)
+```
+
+`broadcast_to` can contain any foreign key field on the model. We set `broadcast_self` to `False` since we don't
+anticipate listening in to updates on a single message, only on the associated `Room`. This is all the set-up on the
+model side of things that we'll need!
+
+### Template Layer
+
+#### Subscribing to a stream
+Next, we can declare that a given page should subscribe to broadcasts for a given model instance. Make sure to
+`{% load turbo_streams_helpers %}` to get access to the template tag in your template.
+
+Make sure to pass the model instance (here, it's passed as `room`) into the template context. Somewhere in the `<body>`,
+we need to add the template tag `{% turbo_stream_from room %}`. That's all we need to do to connect this template, when
+rendered on the front-end, to the stream of updates for a given model instance!
+
+#### Rendering Streams
+Streams need to render HTML of updates on the backend using template partials to send over to the frontend over the wire.
+`turbo-django` at the moment relies on some conventional names for templates and IDs:
+
+- `app_name/model_name.html` should contain the template partial for rendering the model. The context passed in will
+  have the labels `object` as well as `model_name` refer to the instance being rendered. Streams mandates that there's
+  one, top-level element wrapping the HTML. This element should have an ID of the form `<model_name>_<pk>`, where `pk`
+  is the primary key of the instance. The element's ID is necessary to ensure that the Streams update operation, titled
+  `replace`, can predictably find elements on the page.
+  * For our example with the `Message` model, we should create a template called `chat/message.html`, where the top-level
```

Comment:
A concrete code example could help drive the point home.

Reply:
I don't want to crowd the docs with too many examples, do you think it would be fine here to link out to the example app in the directory?

Reply:
We need a cool streamcast of course :)

Reply:
I do love seeing examples in projects I use. Being able to follow along and copy/paste working code gets me running much quicker. IMO a few small HTML snippets wouldn't crowd the docs much.

Reply:
yes, agree. But we could also leave that to another Issue, from my perspective.